lib/dpul_collections_web/content_warnings.ex

```diff
@@ -0,0 +1,141 @@
+defmodule DpulCollectionsWeb.ContentWarnings do
+  use DpulCollectionsWeb, :html
+  use Phoenix.Component
+
+  alias Phoenix.LiveView.JS
+  use Gettext, backend: DpulCollectionsWeb.Gettext
+
+  defmacro __using__(_) do
+    quote do
+      # Add a way to handle the show_images event to all live views.
+      def handle_event(
+            "show_images",
+            %{"id" => id},
+            socket = %{assigns: %{show_images: show_images}}
+          ) do
+        new_show_images =
+          case show_images do
+            nil -> [id]
+            _ -> [id | show_images]
+          end
+
+        {:noreply, socket |> assign(show_images: new_show_images)}
```

Comment:
What use case does storing these in the socket address? If there're not there on navigate? Maybe this is relevant if you click to view on the browse item and then click explore similar for that item? Then its little thumbnail in that action bar was blurred without this?